.github/workflows/build-containers.yml

```diff
@@ -0,0 +1,83 @@
+name: Build & Deploy Docker Containers
+on:
+  # Build new Spack develop containers nightly.
+  schedule:
+    - cron: '0 0 * * *'
+  # Let's also build & tag Spack containers on releases.
+  release:
+    types: [published]
+  # And let's build but not release Spack containers on a PR to develop.
+  pull_request:
```

Comment:
Do we need to build containers on every PR? Or maybe only when a certain set of files are changed that may impact the builds? 🤔

Reply:
No you're right running on a PR isn't necessary since we've got  other unit tests. I think if we want to disable it that totally okay. Mostly I wanted to enable it to have it run on this PR and double check my own submission.

Reply:
@vsoch this kind of got lost in the pile but in your opinion should we drop building the container from all PRs or should we look into building core Spack changes? Or will other unit tests be more than enough?

Reply:
I think it's quite a lot to build for every PR - I would say just have it be a scheduled job at some frequency to build develop.

Reply:
Cool! PR trigger is removed so now we're only building containers nightly and on a Spack release.